config.yml

```diff
@@ -1,584 +1,368 @@
-  - type: assignRegistrant
-    issue: '%actions.repoPR.data.number%'
-  - type: octokit
-    method: pullRequests.createComment
-    owner: '%actions.repoPR.data.head.repo.owner.login%'
-    repo: '%actions.repoPR.data.head.repo.name%'
-    number: '%actions.repoPR.data.number%'
-    commit_id: '%actions.repoPR.data.head.sha%'
-    path: config.yml
-    body: |
-      ```suggestion
-        repo: lab-starter-template
-      ```
-    position: 7
+  - title: Merge your learning objectives
+    description: Merge the pull request breaking down your course topic into smaller learning objectives.
+    event: pull_request.closed
+    link: '{{ repoUrl }}/pull/1'
+    actions:
+    - type: gate
+      every: true
+      gates:
+      - left: '%payload.action%'
+        operator: ===
+        right: closed
+      - left: '%payload.pull_request.merged%'
+    - type: respond
+      with: 03_nice-work.md
+    # Step 4: Choose a template project
+    - type: updateBranchProtection
 
-# 5
-- title: Point to the template repo
-  description: Designate a template repository from which to clone.
-  link: '{{ repoUrl }}/pull/2'
-  event: pull_request.synchronize
-  actions:
-  - type: getFileContents
-    action_id: fileContents
-    filename: 'config.yml'
-  - type: gate
-    left: '/^\s{2}repo: *lab-starter-template(?!#).*$/gm'
-    operator: test
-    right: '%actions.fileContents%'
-    else:
+  - title: Reference the template
+    description: Edit the config file to reference your chosen template repository.
+    event: pull_request.synchronize
+    link: '{{ repoUrl }}/pull/2'
+    actions:
+  # - type: gate # we validate that the repository exists
     - type: respond
-      with: 03.1_e-no-repo.md
-  - type: respond
-    with: 03.0_add-repo-name.md
-  - type: octokit
-    method: pullRequests.createComment
-    owner: '%payload.repository.owner.login%'
-    repo: '%payload.repository.name%'
-    number: '%payload.number%'
-    commit_id: '%payload.pull_request.head.sha%'
-    path: config.yml
-    body: |
-      ```suggestion
-        name: my-sparkles-repo
-      ```
-    position: 6
+      with: 04_good-template.md
+    - type: removeBranchProtection
+    - type: mergeBranch
+      head: template
+    - type: mergeBranch
+      head: master
+      base: map-events
+    - type: deleteBranch
+      branch: template
+  # Step 5: Mapping learning objectives to events and how to choose events that align with the learning objective
+    - type: updateBranchProtection
 
-# 6
-- title: Give the learner's repo a name
-  description: Designate a name for the repository when it is created for the learner.
-  link: '{{ repoUrl }}/pull/2'
-  event: pull_request.synchronize
-  actions:
-  - type: getFileContents
-    action_id: fileContents
-    filename: 'config.yml'
-  - type: gate
-    left: '/^\s{2}name: *\w(?![ \t]*#).*$/gm'
-    operator: test
-    right: '%actions.fileContents%'
-    else:
+  - title: Map learning objectives to events
```

Comment:
@brianamarie on this step, I ran in to a flow error. When I committed the event, I received a bot response of "nice template. I'm going to merge this for you." with a link back to the same PR. I also had a merge conflict even though I had followed the instructions carefully in this part of the course and only changed the fields I was supposed to change in each PR.

Reply:
Hmmm....Interesting! I am going to dig deeper here.